ac-66 get people <category> api

ExtraGravity opened the pull request:
Removed the get all people api for now.
For times sake, if we need to get everyone, we can just get each type
individually.
Right now only certain attribute types are hard coded, but not people.
So if we add an new people category that doesn't use any new attributes,
code doesn't need to change.

Also, I'm pretty sure there's some functional magic that I can do to make this shorter, but I don't know what it is, so please enlighten me :)

ExtraGravity committed:
ac-66 get people <category> api

Removed the get all people api for now.
For times sake, if we need to get everyone, we can just get each type
individually.
Right now only certain attribute types are hard coded, but not people.
So if we add an new people category that doesn't use any new attributes,
code doesn't need to change.

ExtraGravity committed:
ac-66 fix linter errors

ExtraGravity committed:
ac-66 add ramda dependency

beaesguerra commented on src/database/people/index.ts:
```diff
@@ -0,0 +1,115 @@
+const peopleAttributesTable = () => dbConnection()(tableNames.PEOPLE_ATTRIBUTES);
+
+/**
+ * Represents a product transaction, except for it's typeid
```
need to update the comment :)

beaesguerra commented on src/database/people/index.ts:
```diff
@@ -0,0 +1,115 @@
+
+  username?: string;
+  paymentFrequency?: string;
+  notes?: string;
```
should we be returning the type the person is? or nah?

beaesguerra commented on src/database/people/index.ts:
```diff
@@ -0,0 +1,115 @@
+  notes?: string;
+}
+
+interface PeopleAttribute {
```
should this say db after so it's consistent?

beaesguerra commented on src/database/people/index.ts:
```diff
@@ -0,0 +1,115 @@
+}
+
+const builders = {
+  /** Get all product transactions of a certain type */
```
update comment

beaesguerra commented on src/database/people/index.ts:
```diff
@@ -0,0 +1,115 @@
+    .join(tableNames.PEOPLE_CATEGORIES,
+      tableNames.PEOPLE + '.peoplecategoryid',
+      tableNames.PEOPLE_CATEGORIES + '.peoplecategoryid')
+    .where({ peoplecategoryname });
```
ohh hey this is what i was trying to work on before lol :P

beaesguerra commented on src/database/people/index.ts:
```diff
@@ -0,0 +1,115 @@
+    .where({ peoplecategoryname });
+  },
+
+  /** get all attributes and their values for a products certain type */
```
update comments :)

beaesguerra commented on src/database/people/index.ts:
```diff
@@ -0,0 +1,115 @@
+import dbConnection, { tableNames, execute } from '../connection';
+
+const R = require('ramda');
```
I don't think we are supposed to be importing like this. 
Should be like this: `import * as R from 'ramda';`

ExtraGravity replied:
no problem, but what's the difference?

beaesguerra replied:
I'm not sure but Brad says so :P

beaesguerra commented on src/database/people/index.ts:
```diff
@@ -0,0 +1,115 @@
+  const results : Person[] = [];
+
+  people.forEach(function (item) {
+    const isMatchingUuid = R.propEq('personuuid', item.personuuid);
```
hmm it might just be me but I find this syntax difficult to understand. This is just making a function right?

beaesguerra replied:
what's the benefit of using ramda over normal functions?

ExtraGravity replied:
I find this easier to read, this functional approach imo is cleaner than having more loops.

beaesguerra commented on src/routers/people/index.ts:
```diff
@@ -1,166 +1,133 @@
-import createRouter from '@/utilities/functions/createRouter';
```
why did this file change? looks the same?

ExtraGravity replied:
line endings :)

ExtraGravity committed:
Merge branch 'master' into enoch/ac-66-get-people-type-api

ExtraGravity committed:
ac-66 respond to bea's pr #67

Change how ramda is imported, included type definitions.
Updated comments to refer ot people instead of products.
Add returning the type of person.
Add consistency for referring to database interfaces suffixed with db.
